sr_error_reporter/include/sr_error_reporter/UnderactuationErrorReporter.hpp

```diff
@@ -0,0 +1,116 @@
+/*
+* Copyright 2021 Shadow Robot Company Ltd.
+*
+* This program is free software: you can redistribute it and/or modify it
+* under the terms of the GNU General Public License as published by the Free
+* Software Foundation version 2 of the License.
+*
+* This program is distributed in the hope that it will be useful, but WITHOUT
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+* more details.
+*
+* You should have received a copy of the GNU General Public License along
+* with this program. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef SR_INTERFACE_SR_ERROR_REPORTER_SRC_UNDERACTUATIONERRORREPORTER_HPP_
+#define SR_INTERFACE_SR_ERROR_REPORTER_SRC_UNDERACTUATIONERRORREPORTER_HPP_
+
+#include <map>
+#include <set>
+#include <string>
+#include <vector>
+
+#include <geometry_msgs/Transform.h>
+#include <moveit/robot_model_loader/robot_model_loader.h>
+#include <moveit/robot_model/robot_model.h>
+#include <moveit/robot_state/robot_state.h>
+#include <ros/ros.h>
+#include <sensor_msgs/JointState.h>
+#include <trajectory_msgs/JointTrajectory.h>
+
+class UnderactuationErrorReporter
+{
+ public:
+  explicit UnderactuationErrorReporter(ros::NodeHandle& node_handle);
+ private:
+  ros::NodeHandle& node_handle_;
+  robot_model_loader::RobotModelLoaderPtr robot_model_loader_;
+  moveit::core::RobotStatePtr robot_state_;
+  ros::Subscriber joints_subsriber_, trajectory_subsriber_left_, trajectory_subsriber_right_;
+  std::map<std::string, ros::Publisher> error_publishers_;
+
+  /**
+   * MoveIt link name to radians. From /joint_states topic.
+   */
+  std::map<std::string, double> actual_joint_angles_;
+
+  /**
+   * MoveIt link name to radians. From /[lh|rh]_trajectory_controller/command topic.
+   */
+  std::map<std::string, double> desired_joint_angles_;
+
+  std::map<std::string, geometry_msgs::Transform> actual_tip_transforms_;
+  std::map<std::string, geometry_msgs::Transform> desired_tip_transforms_;
+
+  std::set<std::string> sides_;
+
+  std::map<std::string, std::string> include_fingers_ =
```

Comment:
we need to be careful when setting this in launch file. Lite hand does not have ring finger and little finger, not sure if we are interested in checking the error there, just heads up.

Reply:
It might be a bit confusing but this map purpose it to exclude thumb and wrist joints. If ring finger and little finger are not present in /joint_states or /[lh|rh]_trajectory_controller/command they will be ignored too. So errors for ring and little fingers will be published only if lite hand publishes commands and joint states for them. Does it?